source/simulator/assembly.cc

```diff
@@ -1286,6 +1286,20 @@ namespace aspect
           return residuals;
         }
 
+        double
+        dg_approximation_to_face_measure(const typename DoFHandler<2>::face_iterator face) const
+        {
```

Comment:
finally, I think the argument type should be `Triangulation<2>::face_iterator`. The `typename` is wrong here in C++03 and unnecessary in C++11. There is a conversion operator from the DoFHandler iterator to the Triangulation iterator.